test/features/support/ui_authorization_auth_code.js

```diff
@@ -0,0 +1,133 @@
+const pactum = require("pactum");
+const { Given, When, Then, Before, After } = require("@cucumber/cucumber");
+const {
+  localhost,
+  validTransactionId,
+  requestTime,
+} = require("./helpers/helpers");
+
+let specUIAuthCode;
+let invalidTransactionId;
+
+const baseUrl = `${localhost}authorization/auth-code`;
+
+const responseSchema = {
+  type: "object",
+  properties: {
+    responseTime: {
+      type: "string",
+    },
+    response: {
+      type: "object",
+      properties: {
+        code: {
+          type: "string",
+        },
+        redirectUri: {
+          type: "string",
+        },
+        nonce: {
+          type: "string",
+        },
+        state: {
+          type: "string",
+        },
+      },
+      additionalProperties: false,
+    },
+    errors: {},
+  },
+  additionalProperties: false,
+};
+
+const requestFunction = (transactionId) =>
+  specUIAuthCode.post(baseUrl).withBody({
+    requestTime: requestTime,
+    request: {
+      transactionId: transactionId,
+      permittedAuthorizeScopes: ["string"],
+      acceptedClaims: ["string"],
+    },
+  });
+
+const responseValidator = async (withErrors = false) => {
+  if (!withErrors) {
+    responseSchema.properties.errors = { type: "array" };
+  } else {
+    responseSchema.properties.errors = {
+      type: "array",
+      items: {
+        type: "object",
+        properties: {
+          errorCode: {
+            type: "string",
+            enum: ["invalid_request", "invalid_transaction"],
+          },
+          errorMessage: {
+            type: "string",
+          },
+        },
+        additionalProperties: false,
+      },
+    };
+  }
+
+  specUIAuthCode.expectStatus(200).expectJsonSchema(responseSchema);
```

Comment:
Please take a closer look at the schema.
Even if you remove ```transactionId``` property from the request, all the test will finish successfully.